### ISSUE TITLE
Implement `gelsd` backend for `linalg.lstsq` via `linalg.svd`

### ISSUE THREAD

**Issue by Lezcano** [pytorch/pytorch#72711]
### 🚀 The feature, motivation and pitch

As per title.

**Comment by Liquidtoss** [comment]
Can I attempt to implement this?

**Comment by Lezcano** [comment]
Absolutely. You might want to have a look at a function like `matrix_rank` which is also implemented in terms of `at::linalg_svd` for inspiration.
Please feel free to contact me here or in PyTorch's slack if you need some guidance on how to do this.
